api/src/Service/CCService.php

```diff
@@ -13,23 +19,201 @@ class CCService
     private CommonGroundService $commonGroundService;
     private ParameterBagInterface $parameterBag;
     private EAVService $eavService;
+    private WRCService $wrcService;
+    private EDUService $eduService;
+    private UcService $ucService;
+    private MrcService $mrcService;
 
-    public function __construct(EntityManagerInterface $entityManager, CommonGroundService $commonGroundService, ParameterBagInterface $parameterBag, EAVService $eavService)
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        CommonGroundService $commonGroundService,
+        ParameterBagInterface $parameterBag,
+        EAVService $eavService,
+        WRCService $wrcService,
+        EDUService $eduService,
+        UcService $ucService,
+        MrcService $mrcService
+    )
     {
         $this->entityManager = $entityManager;
         $this->commonGroundService = $commonGroundService;
         $this->parameterBag = $parameterBag;
         $this->eavService = $eavService;
+        $this->wrcService = $wrcService;
+        $this->eduService = $eduService;
+        $this->ucService = $ucService;
+        $this->mrcService = $mrcService;
     }
 
-    public function saveOrganization(array $body, $type = null, $sourceOrgurl = null){
-        if (isset($type)) $body['type'] = $type;
-        if (isset($sourceOrgurl)) $body['sourceOrganization'];
-            return $this->commonGroundService->saveResource($body, ['component' => 'cc', 'type' => 'organization']);
+    public function createOrganization(array $organizationArray, $type)
+    {
+        $wrcOrganization = $this->wrcService->createOrganization($organizationArray);
+        $address = [
+            'name' => 'Address of '.$organizationArray['name'],
+            'street' => $organizationArray['address']['street'],
+            'houseNumber' => $organizationArray['address']['houseNumber'],
+            'houseNumberSuffix' => $organizationArray['address']['postalCode'],
+            'postalCode' => $organizationArray['address']['postalCode'],
+            'locality' => $organizationArray['address']['locality'],
+        ];
```

Comment:
To create clean, concise functions you _could_ think of offloading the creation of these arrays to a separate function, but that might be a bit extreme here.